### PR TITLE
tmp: editor package with vite

### PR DESCRIPTION
**Pull request by elbotho**
this is the first step: export editor as react component

### what works: 
- editor and tools to customize can be imported and used in other react repo
- tailwind css styles are exported and can be imported from package
- important types are included

### what does not work / is not included here:
- assets (icons, katex font)
- using workspaces
- limit dependencies

### example usage from other repo

``` ts
import {
  RowsStaticRenderer,
  TextStaticRenderer,
  editorPlugins,
  editorRenderers,
  instanceDataDe,
  loggedInDataDe,
  SerloEditor,
  EditorPluginType,
  createTextPlugin,
  createRowsPlugin,
  MathElement,
  StaticMath,
} from '@serlo/editor'
import {
  InstanceData,
  LoggedInData,
} from '@serlo/editor/dist/src/data-types'
import { LinkRenderer } from '@serlo/editor/dist/src/serlo-editor/plugin/helpers/editor-renderer'
import { ComponentProps } from 'react'

export default function Demo() {
  editorPlugins.init([
    {
      type: EditorPluginType.Text,
      plugin: createTextPlugin({}),
      visibleInSuggestions: true,
      icon: <>text</>,
    },
    { type: EditorPluginType.Rows, plugin: createRowsPlugin() },
  ])

  editorRenderers.init({
    pluginRenderers: [
      { type: EditorPluginType.Rows, renderer: RowsStaticRenderer },
      { type: EditorPluginType.Text, renderer: TextStaticRenderer },
    ],
    mathRenderer: (element: MathElement) => <StaticMath {...element} />,
    linkRenderer: ({ href, children }: ComponentProps<LinkRenderer>) => {
      return <a href={href}>{children}</a>
    },
  })

  return (
    <SerloEditor
      instanceData={instanceDataDe as InstanceData}
      loggedInData={loggedInDataDe as LoggedInData}
    />
  )
}

```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
+    "dist"
+  ],
   "scripts": {
     "_eslint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
     "_prettier": "prettier .",
     "analyze": "cross-env ANALYZE=true yarn build",
     "build": "npm-run-all build:prisma build:next",
     "build:next": "next build",
     "build:prisma": "yarn prisma generate",
+    "build-editor": "tsc && vite build && tailwindcss -o dist/editor-tailwind.css --minify",
```
Comment (kulla) [inline package.json]: @elbotho: Why do you run `tsc` here?

Comment (elbotho) [reply package.json]: It's mostly how vite suggests it.

**Comment by elbotho** [comment]
closing this because it should slowly but surely move into https://github.com/serlo/frontend/pull/3183

**Comment by kulla** [comment]
> closing this because it should slowly but surely move into #3183

Keep it open for now, I am trying some things -> I will close it... :smile:

**Comment by kulla** [comment]
## ToDos to fix TS errors

- [x] 1  src/frontend/editor.tsx:41
- [x] 2  src/frontend/plugins/create-renderers.tsx:6
- [x] 2  src/frontend/plugins/edusharing-asset/editor.tsx:4
- [x] 10  src/frontend/plugins/serlo-injection/editor.tsx:5
- [x] 1  src/frontend/plugins/serlo-injection/index.ts:5
- [x] 1  src/frontend/plugins/serlo-injection/static.tsx:9
- [x] "Error: could not find react-redux context value; please ensure the component is wrapped in a <Provider>"

**Comment by kulla** [comment]
Work will be continued in https://github.com/serlo/frontend/pull/3220

**Pull request closed without merging**